src/parachain/system.ts

```diff
@@ -60,6 +66,13 @@ export class DefaultSystemAPI {
         return unsub;
     }
 
+    async subscribeToCurrentBlockHeads(callback: (blockHeader: Header) => void): Promise<() => void> {
+        const unsub = await this.api.rpc.chain.subscribeAllHeads((head) => {
+            callback(head);
```

Comment:
If the callback is async, this will not await. That might not be a problem since the async calls will probably resolve before a new block comes in, but it's something worth considering as it might create race conditions.

Reply:
Thanks for pointing out. Good catch.
 
Also a consideration for `subscribeToFinalizedBlockHeads` (which I've shamelessly copy/pasta'd for this one)
https://github.com/interlay/interbtc-api/blob/master/src/parachain/system.ts#L56